src/types/shortcuts.ts

```diff
@@ -1,25 +1,34 @@
 import { z, ZodObject, ZodRawShape } from 'zod'
 import { NetworkId } from './networkId'
 
+export type ShortcutCategory = 'claim' | 'deposit' | 'withdraw'
+
 export interface ShortcutsHook {
   getShortcutDefinitions(
     networkId: NetworkId,
     address?: string,
-  ): Promise<ShortcutDefinition<any>[]>
+  ): Promise<ShortcutDefinition<ShortcutCategory, any>[]>
 }
 
-export interface ShortcutDefinition<TriggerInputShape extends ZodRawShape> {
+type TriggerInputShape<Category> = Category extends 'deposit' | 'withdraw'
+  ? ZodRawShape & { amount: z.ZodString }
```

Comment:
I feel it's slightly surprising only amount is required here.
An amount without a tokenId or address is not very useful.

Is there a problem adding the other required fields we want for `deposit` and `withdraw`?

Reply:
the reason I left it out was because it was the only thing that needs to be set by the client and everything else can come from shortcutTriggerArgs, but I agree that it's cleaner with having token id / address + amount together. 

I wonder if we should just do the input as an array of tokenIds + decimal amount? If we do a single token here and if we keep the address + decimals format right now, changing to tokenId would be tricky to maintain backwards compatibility. Guess another option to not require decimals is to directly take the amount in smallest unit, but we still need token address, which we have to lookup using the tokenId, at which point we can get decimals too. So probably fine to just require tokenId + decimal amount and then do the lookup. What do you think?

Reply:
Chatted offline and landed on a tokens format with tokenId and amount. As a temp workaround, address and decimals are passed via shortcutTriggerArgs, until we can look it up using tokenId